opm/parser/eclipse/EclipseState/Schedule/Well/Well.hpp

```diff
@@ -383,6 +384,35 @@ class Well {
           const UnitSystem& unit_system,
           double udq_undefined);
 
+    Well(const std::string& wname,
+         const std::string& gname,
+         std::size_t init_step,
+         std::size_t insert_index,
+         int headI,
+         int headJ,
+         double ref_depth,
+         const Phase& phase_arg,
+         Connection::Order ordering,
+         const UnitSystem& unit_system,
+         double udq_undefined,
+         Status status,
+         double drainageRadius,
+         bool allowCrossFlow,
+         bool automaticShutIn,
+         bool isProducer,
+         const WellGuideRate& guideRate,
+         double efficiencyFactor,
+         double solventFraction,
+         bool prediction_mode,
+         std::shared_ptr<const WellEconProductionLimits> econLimits,
```

Comment:
If possible I would prefer to let the use of `std::shared_ptr<>` be an implementation detail - not exposed in the API.

Reply:
well, i'm not really using shared_ptr to reflect internal semantics of the Well class here. I need to move a pointer between instances and using a shared_ptr is a convenient and safe way to do so.

Reply:
OK - thanks for the clarification.